packages/xrpl/src/client/partialPayment.ts

```diff
@@ -82,14 +85,17 @@ function txEntryHasPartialPayment(response: TransactionEntryResponse): boolean {
 
 function accountTxHasPartialPayment(response: AccountTxResponse): boolean {
   const { transactions } = response.result
-  const foo = transactions.some((tx) => isPartialPayment(tx.tx, tx.meta))
+  const foo = transactions.some((tx) =>
+    // API v2 returns tx_json, API v1 returns tx
+    isPartialPayment(tx.tx_json ?? tx.tx, tx.meta),
```

Comment:
This line isn't accurate. If you mouseover `tx.tx` it's showing as always `undefined` because `AccountTxresponse` defaults to `AccountResponseBase<2>`. Your code "works" but the type checking is wrong.

You need to make `accountTxHasPartialPayment` a template function that properly passes in the type of the API version and then `AccountTxResponse` also needs to be made generic in order to pass the version to that which is eventually passed to `AccountResponseBase` which gives you the proper typing (1 or 2)